### ISSUE TITLE
MODS schema validation failures

### ISSUE THREAD

**Issue by wgilling** [asulibraries/islandora-repo#364]
Some of these problems may have happened before and some may now be happening because more data is actually being encoded. The resultant MODS for any object can be validated (using the schemaLocation schema) here: https://www.freeformatter.com/xml-validator-xsd.html ... current list of validation issues are:

> Cvc-enumeration-valid: Value 'corporate_body' Is Not Facet-valid With Respect To Enumeration '[personal, Corporate, Conference, Family]'. It Must Be A Value From The Enumeration., Line '2', Column '78'.
> Cvc-attribute.3: The Value 'corporate_body' Of Attribute 'type' On Element 'name' Is Not Valid With Respect To Its Type, '#AnonType_typenameDefinition'., Line '2', Column '78'.
> Cvc-complex-type.2.4.a: Invalid Content Was Found Starting With Element 'roleTerm'. One Of '{"http://www.loc.gov/mods/v3":namePart, "http://www.loc.gov/mods/v3":displayForm, "http://www.loc.gov/mods/v3":affiliation, "http://www.loc.gov/mods/v3":role, "http://www.loc.gov/mods/v3":description, "http://www.loc.gov/mods/v3":nameIdentifier, "http://www.loc.gov/mods/v3":alternativeName, "http://www.loc.gov/mods/v3":etal}' Is Expected., Line '2', Column '100'.
> Cvc-complex-type.2.4.a: Invalid Content Was Found Starting With Element 'roleTerm'. One Of '{"http://www.loc.gov/mods/v3":namePart, "http://www.loc.gov/mods/v3":displayForm, "http://www.loc.gov/mods/v3":affiliation, "http://www.loc.gov/mods/v3":role, "http://www.loc.gov/mods/v3":description, "http://www.loc.gov/mods/v3":nameIdentifier, "http://www.loc.gov/mods/v3":alternativeName, "http://www.loc.gov/mods/v3":etal}' Is Expected., Line '2', Column '257'.
> Cvc-complex-type.2.4.a: Invalid Content Was Found Starting With Element 'roleTerm'. One Of '{"http://www.loc.gov/mods/v3":namePart, "http://www.loc.gov/mods/v3":displayForm, "http://www.loc.gov/mods/v3":affiliation, "http://www.loc.gov/mods/v3":role, "http://www.loc.gov/mods/v3":description, "http://www.loc.gov/mods/v3":nameIdentifier, "http://www.loc.gov/mods/v3":alternativeName, "http://www.loc.gov/mods/v3":etal}' Is Expected., Line '2', Column '370'.
> Cvc-complex-type.2.4.b: The Content Of Element 'language' Is Not Complete. One Of '{"http://www.loc.gov/mods/v3":languageTerm}' Is Expected., Line '2', Column '485'.
> Cvc-datatype-valid.1.2.1: 'http:' Is Not A Valid Value For 'anyURI'., Line '2', Column '885'.
> Cvc-attribute.3: The Value 'http:' Of Attribute 'valueURI' On Element 'genre' Is Not Valid With Respect To Its Type, 'anyURI'., Line '2', Column '885'.
> Cvc-complex-type.3.2.2: Attribute 'src' Is Not Allowed To Appear In Element 'recordIdentifier'., Line '2', Column '1331'.
> Cvc-complex-type.2.3: Element 'name' Cannot Have Character [children], Because The Type's Content Type Is Element-only., Line '2', Column '2455'.

The peer-reviewed and open access boolean fields need to be handled differently because we want to skip the element completely in cases where they are == 0. Suggested rework to the script and ModsEncoder to handle a couple new config fields such as:
```
field_peer_reviewed:
  '_top': genre
  '@authority': "getty"
  "@valueURI": "http://vocab.getty.edu/page/aat/300380321"
  '#': field_peer_reviewed
  '_skip_zero': true
  '_string_val': "peer reviewed"
field_open_access:
  '_top': genre
  '@authority': "local"
  '@valueURI': "https://repository.lib.asu.edu/taxonomy/term/1"
  '#': field_open_access
  '_skip_zero': true
  '_string_val': "open access"
```

**Comment by wgilling** [comment]
This is the adjusted MODS file that was used for the validation test in the description of the issue. It was adjusted so that the outer modsCollection wrapper was removed by the attributes of modsCollection were put on to the mods element.
[mods.xml.txt](https://github.com/asulibraries/islandora-repo/files/6250563/mods.xml.txt)

**Comment by elizoller** [comment]
is this a duplicate of #345 ? or supercedes 345?

**Comment by wgilling** [comment]
I believe that this issue came after #345 yet an entirely fresh mods XML schema validation will need to be done because there may have been xml nodes that did not exhibit any issues before and possibly (less likely) there may be issues that were fixed via other commits.

**Comment by wgilling** [comment]
The latest pass on the MODS xml schema validation is:

- Cvc-enumeration-valid: Value 'corporate_body' Is Not Facet-valid With Respect To Enumeration '[personal, Corporate, Conference, Family]'. It Must Be A Value From The Enumeration., Line '1', Column '441'.
- Cvc-attribute.3: The Value 'corporate_body' Of Attribute 'type' On Element 'name' Is Not Valid With Respect To Its Type, '#AnonType_typenameDefinition'., Line '1', Column '441'.

```
   <name type="corporate_body" authority="lcnaf" valueURI="http://id.loc.gov/authorities/names/no00073503">
      <role>
         <roleTerm type="code">pbl</roleTerm>
      </role>
      <namePart><![CDATA[Francis, Day & Hunter]]></namePart>
   </name>
```

This element is created in the asu_mods.asu_repository_item.yml script by setting the type attribute to the taxonomy field's referenced bundle:
```
field_linked_agent:
  '_top': name
  '@type': bundle
  '@authority': field_authority_link/source
  '@valueURI': field_authority_link/uri
  role:
    roleTerm:
      '@type': 'code'
      '#': rel_type
  namePart: name
```

**Comment by elizoller** [comment]
So we need to translate the bundle machine name into what MODS expects

**Comment by elizoller** [comment]
closed with #420
